haystack/components/readers/extractive.py

```diff
@@ -299,21 +308,122 @@ def _nest_answers(
                 answer.query = queries[query_id]
                 current_answers.append(answer)
                 i += 1
-            current_answers = sorted(current_answers, key=lambda answer: answer.score, reverse=True)
+            current_answers = sorted(current_answers, key=lambda ans: ans.score, reverse=True)
+            current_answers = self.deduplicate_by_overlap(current_answers, overlap_threshold=overlap_threshold)
             current_answers = current_answers[:top_k]
             if no_answer:
                 no_answer_score = math.prod(1 - answer.score for answer in current_answers)
                 answer_ = ExtractedAnswer(
                     data=None, query=queries[query_id], meta={}, document=None, score=no_answer_score
                 )
                 current_answers.append(answer_)
-            current_answers = sorted(current_answers, key=lambda answer: answer.score, reverse=True)
+            current_answers = sorted(current_answers, key=lambda ans: ans.score, reverse=True)
```

Comment:
Why this change? (same goes for line 302)

Reply:
Complaints from pycharm for `Shadows name 'answer' from outer scope`. I know it shouldn't really matter, but I thought it would still be worth changing since I don't think it affects readability and removes the warning. Should I change back?

Reply:
Ah didn't notice, let's keep it like this 👍